src/module_2/module_2.md

```diff
@@ -0,0 +1,3512 @@
+
+Information about users
+
+
+```python
+print(df_users.shape)
+df_users.head()
+```
+
+    (4983, 10)
+
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>user_id</th>
+      <th>user_segment</th>
+      <th>user_nuts1</th>
+      <th>first_ordered_at</th>
+      <th>customer_cohort_month</th>
+      <th>count_people</th>
+      <th>count_adults</th>
+      <th>count_children</th>
+      <th>count_babies</th>
+      <th>count_pets</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>2160</th>
+      <td>0e823a42e107461379e5b5613b7aa00537a72e1b0eaa7a...</td>
+      <td>Top Up</td>
+      <td>UKH</td>
+      <td>2021-05-08 13:33:49</td>
+      <td>2021-05-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+    <tr>
+      <th>1123</th>
+      <td>15768ced9bed648f745a7aa566a8895f7a73b9a47c1d4f...</td>
+      <td>Top Up</td>
+      <td>UKJ</td>
+      <td>2021-11-17 16:30:20</td>
+      <td>2021-11-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+    <tr>
+      <th>1958</th>
+      <td>33e0cb6eacea0775e34adbaa2c1dec16b9d6484e6b9324...</td>
+      <td>Top Up</td>
+      <td>UKD</td>
+      <td>2022-03-09 23:12:25</td>
+      <td>2022-03-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+    <tr>
+      <th>675</th>
+      <td>57ca7591dc79825df0cecc4836a58e6062454555c86c35...</td>
+      <td>Top Up</td>
+      <td>UKI</td>
+      <td>2021-04-23 16:29:02</td>
+      <td>2021-04-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+    <tr>
+      <th>4694</th>
+      <td>085d8e598139ce6fc9f75d9de97960fa9e1457b409ec00...</td>
+      <td>Top Up</td>
+      <td>UKJ</td>
+      <td>2021-11-02 13:50:06</td>
+      <td>2021-11-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+Assess general info about NANs
+
+
+```python
+def assess_NA(data):
+    """
+    Returns a pandas dataframe denoting the total number of NA
+    values and the percentage of NA values in each column.
+    The column names are noted on the index.
+    """
+    # pandas series denoting features and the sum of their null values
+    null_sum = data.isnull().sum()# instantiate columns for missing data
+    total = null_sum.sort_values(ascending=False)
+    percent = ( ((null_sum / len(data.index))*100).round(2) ).sort_values(ascending=False)
```

Comment:
Very nit pick: I assume you are using the round for the print, to avoid a lot of decimals in the console. If this is the case, the one in charge of this logic should be the print itself. Have a look at: https://stackoverflow.com/questions/43217916/pandas-data-precision